account_setup.py

```diff
@@ -0,0 +1,348 @@
+'''
+    Getting Started AZTK script
+'''
+import time
+import uuid
+import yaml
+from azure.common import credentials
+from azure.graphrbac import GraphRbacManagementClient
+from azure.graphrbac.models import ApplicationCreateParameters, PasswordCredential, ServicePrincipalCreateParameters
+from azure.graphrbac.models.graph_error import GraphErrorException
+from azure.mgmt.authorization import AuthorizationManagementClient
+from azure.mgmt.batch import BatchManagementClient
+from azure.mgmt.batch.models import AutoStorageBaseProperties, BatchAccountCreateParameters
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.network.models import AddressSpace, Subnet, VirtualNetwork
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.storage import StorageManagementClient
+from azure.mgmt.storage.models import Kind, Sku, SkuName, StorageAccountCreateParameters
+from azure.mgmt.subscription import SubscriptionClient
+from datetime import datetime, timezone
+from msrestazure.azure_cloud import AZURE_PUBLIC_CLOUD
+from msrestazure.azure_exceptions import CloudError
+
+
+class AccountSetupError(Exception):
+    pass
+
+
+class DefaultSettings():
+    resource_group = 'aztk'
+    storage_account = 'aztkstorage'
+    batch_account = 'aztkbatch'
+    virtual_network_name = "aztkvnet"
+    subnet_name = 'aztksubnet'
+    application_name = 'aztkapplication'
+    application_credential_name = 'aztkapplicationcredential'
+    service_principal = 'aztksp'
+    region = 'westus'
+
+
+def create_resource_group(credentials, subscription_id, **kwargs):
+    """
+        Create a resource group
+        :param credentials: msrestazure.azure_active_directory.AdalAuthentication
+        :param subscription_id: str
+        :param **resource_group: str
+        :param **region: str
+    """
+    resource_client = ResourceManagementClient(credentials, subscription_id)
+    resource_client.resource_groups.list()
+    resource_group = resource_client.resource_groups.create_or_update(
+        resource_group_name=kwargs.get("resource_group", DefaultSettings.resource_group),
+        parameters={
+            'location': kwargs.get("region", DefaultSettings.region),
+        }
+    )
```

Comment:
Will this fail w/ a good error msg is the user provides bad inputs? E.g. region == westuss

Reply:
Right now this will just throw whatever error the ResourceManagementClient throws. I think we should probably catch this error, print a link to docs on azure regions, do a 3 attempt retry loop.

Reply:
This is the current error: 
```
Traceback (most recent call last):
  File "account_setup.py", line 336, in <module>
    resource_group_id = create_resource_group(creds, subscription_id, **kwargs)
  File "account_setup.py", line 79, in create_resource_group
    'location': kwargs.get("region", DefaultSettings.region),
  File "/home/jake/.local/lib/python3.5/site-packages/azure/mgmt/resource/resources/v2017_05_10/operations/resource_groups_operations.
py", line 147, in create_or_update
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: LocationNotAvailableForResourceGroup
Message: The provided location 'westuss' is not available for resource group. List of available regions is 'centralus,eastasia,southea
stasia,eastus,eastus2,westus,westus2,northcentralus,southcentralus,westcentralus,northeurope,westeurope,japaneast,japanwest,brazilsout
h,australiasoutheast,australiaeast,westindia,southindia,centralindia,canadacentral,canadaeast,uksouth,ukwest,koreacentral,koreasouth,f
rancecentral,eastus2euap,centraluseuap'.
```